public/app/plugins/datasource/tempo/QueryEditor/QueryField.tsx

```diff
@@ -123,12 +123,20 @@ class TempoQueryFieldComponent extends React.PureComponent<Props, State> {
             <RadioButtonGroup<TempoQueryType>
               options={queryTypeOptions}
               value={query.queryType}
-              onChange={(v) =>
+              onChange={(v) => {
+                // Run clear query to clear results
+                onChange({
+                  ...query,
+                  queryType: 'clear',
+                });
+                this.props.onRunQuery();
```

Comment:
What do you think about creating method `onClearResults` on this component that would wrap these functions? I think that it would make it more readable.